ngraph/test/files/paddlepaddle/gen_scripts/generate_pool2d.py

```diff
@@ -0,0 +1,232 @@
+        exe.run(pdpd.static.default_startup_program())
+
+        outs = exe.run(
+            feed={'x': x},
+            fetch_list=[out])
+
+        saveModel(name, exe, feedkeys=['x'], fetchlist=[out], inputs=[x], outputs=[outs[0]], target_dir=sys.argv[1])
+
+    return outs[0]
+
+def adaptive_pool2d(name : str, x, attrs : dict):
+    import paddle as pdpd
+    pdpd.enable_static()
+
+    with pdpd.static.program_guard(pdpd.static.Program(), pdpd.static.Program()):
+        node_x = pdpd.static.data(name='x', shape=x.shape, dtype=data_type)
+        out = pdpd.fluid.layers.adaptive_pool2d(
+            input=node_x,
+            pool_size=attrs['pool_size'],
+            pool_type=attrs['pool_type'],
+            require_index=attrs['require_index'])
+
+        cpu = pdpd.static.cpu_places(1)
+        exe = pdpd.static.Executor(cpu[0])
+        # startup program will call initializer to initialize the parameters.
+        exe.run(pdpd.static.default_startup_program())
+
+        outs = exe.run(
+            feed={'x': x},
+            fetch_list=[out])
+
+        saveModel(name, exe, feedkeys=['x'], fetchlist=[out], inputs=[x], outputs=[outs[0]], target_dir=sys.argv[1])
+
+    return outs[0]
+
+def main():
+    N, C, H, W = 2, 3, 4, 4
+    data = np.arange(N*C*H*W).astype(data_type)
+    data_NCHW = data.reshape(N, C, H, W)
+    data_NHWC = data.reshape(N, H, W, C)
+    #print(data_NCHW, data_NCHW.shape)
+
+    pooling_types = ['max', 'avg']
+
+    # pool2d
+    for i, pooling_type in enumerate(pooling_types):
+        # example 1:
+        # ceil_mode = False
+        pdpd_attrs = {
+            # input=data_NCHW, # shape: [2, 3, 8, 8]
+            'pool_size' : [3,3],
+            'pool_type' : pooling_type,
+            'pool_stride' : [3,3],
+            'pool_padding' : [2,1], # it is same as pool_padding = [2,2,1,1]
+            'global_pooling' : False,
+            'ceil_mode' : False,
+            'exclusive' : True,
+            'data_format' : "NCHW"
+        }
+        # shape of out_1: [2, 3, 4, 3]
+        pool2d(pooling_type+'Pool_test1', data_NCHW, pdpd_attrs)
+
+        # Cecilia: there is a bug of PaddlePaddle in this case.
+        # example 2:
+        # ceil_mode = True (different from example 1)
+        pdpd_attrs = {
+            #input=data_NCHW,
+            'pool_size':[3,3],
+            'pool_type' : pooling_type,
+            'pool_stride' : [3,3],
+            'pool_padding':[[0,0], [0,0], [2,2], [1,1]], # it is same as pool_padding = [2,2,1,1]
+            'global_pooling':False,
+            'ceil_mode':True,
+            'exclusive':True,
+            'data_format':"NCHW"
+        }
+        # shape of out_2: [2, 3, 4, 4] which is different from out_1
+        pool2d(pooling_type+'Pool_test2', data_NCHW, pdpd_attrs)
+
+        # example 3:
+        # pool_padding = "SAME" (different from example 1)
+        pdpd_attrs = {
+            #input=data_NCHW,
+            'pool_size':[3,3],
```

Comment:
[0,0] looks valid value for PDPD, but out tests fail when I set [0,0] here

Reply:
@ceciliapeng2011 please check this situation

Reply:
@nosovmik Our unit test fails because our Pool Op has an ASSERT that the kernel `Check 'window_dilated_dim > 0'` I think kernel [0, 0] is actually not a valid input, but PDPD somehow doesn't handle this correctly. Can we add a ASSERT here to avoid this corner issue ?